src/ralph/models/edx/problem_interaction/statements.py

```diff
@@ -0,0 +1,301 @@
+"""Problem interaction events model definitions"""
+
+from typing import List, Literal, Union
+
+from pydantic import Json
+
+from ralph.models.selector import selector
+
+from ..browser import BaseBrowserModel
+from ..server import BaseServerModel
+from .fields.events import (
+    EdxProblemHintDemandhintDisplayedEventField,
+    EdxProblemHintFeedbackDisplayedEventField,
+    ProblemCheckEventField,
+    ProblemCheckFailEventField,
+    ProblemGradedEventField,
+    ProblemRescoreEventField,
+    ProblemRescoreFailEventField,
+    ResetProblemEventField,
+    ResetProblemFailEventField,
+    SaveProblemFailEventField,
+    SaveProblemSuccessEventField,
+    ShowAnswerEventField,
+    UIProblemCheckEventField,
+    UIProblemResetEventField,
+    UIProblemShowEventField,
+)
+
+
+class EdxProblemHintDemandhintDisplayed(BaseServerModel):
```

Comment:
Thanks for this crucial information. I did not pay attention to it. 
I would prefer to use an optional module field rather than creating an additional model.